jcl/src/java.base/share/classes/openj9/internal/criu/security/CRIUConfigurator.java

```diff
@@ -101,6 +152,52 @@ public static void setCRIURestoreMode() {
 					System.out.println(provider + " restored.");
 				}
 			}
+			// If any algorithms are available from when we took a checkpoint in our algorithm cache
+			// we can now allow callers to get values from this cache.
+			if (cachedAlgorithms.size() > 0) {
+				if (debug) {
+					System.out.println("Algorithm cache enabled.");
+				}
+			    isCacheEligible = true;
+			}
+		}
+	}
+
+	/**
+	 * Gets algorithms from the checkpointed cache of algorithms.
+	 *
+	 * @param serviceName
```

Comment:
Please provide documentation for the parameter and add `@return`.
The javadoc doesn't say this method returns `null` even though it might: either document that or make it impossible.

Reply:
I don't think we can allow this to return `null`, given its use in `Security.getAlgorithms(String)`.

This should return an unmodifiable set.

The documentation for `Security.getAlgorithms(String)` says the service name "is case-insensitive": the code here doesn't treat it that way.

Reply:
Added javadoc for the `@return` value, and better javadoc for `@serviceName`. 

I agree that we do not want to return `null` from this method. The code was updated to return an empty collection instead of `null` when the value was not found in the cache.

The code was also changed to return an unmodifiable collection in the case of something being returned from the cache, the javadoc was updated accordingly to state this.